### PR TITLE
Simplify project cache cleanup

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
To hopefully remove a deadlock we're seeing in this code in CI.

### CODE REVIEW

File: src/EditorFeatures/Core/Workspaces/ProjectCacheServiceFactory.cs [hunk 1]
```diff
@@ -27,21 +25,10 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
...
-            var service = new ProjectCacheService(workspaceServices.Workspace, s_implicitCacheTimeout);
-
-            // Also clear the cache when the solution is cleared or removed.
-            workspaceServices.Workspace.WorkspaceChanged += (s, e) =>
-            {
-                if (e.Kind is WorkspaceChangeKind.SolutionCleared or WorkspaceChangeKind.SolutionRemoved)
-                {
-                    service.ClearImplicitCache();
-                }
-            };
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Workspaces/ProjectCacheServiceFactory.cs:42]: moved this common logic into the ProjectCacheService

File: src/Features/Core/Portable/Workspace/ProjectCacheService.SimpleMRUCache.cs [hunk 2]
```diff
@@ -61,79 +55,20 @@ public void Touch(object instance)
...
-                public Node(object data, DateTime lastTouched)
+                public Node(object? data, DateTime lastTouched)
                 {
                     Data = data;
                     LastTouched = lastTouched;
                 }
             }
         }
-
-        private class ImplicitCacheMonitor : IdleProcessor
```
Comment (dibarbet) [inline src/Features/Core/Portable/Workspace/ProjectCacheService.SimpleMRUCache.cs:91]: bye!👋

**Review by dibarbet — APPROVED** [review]
rps val - https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6683686&view=results

**Comment by CyrusNajmabadi** [comment]
Insertion: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6686339&view=results

**Comment by CyrusNajmabadi** [comment]
Merging in.  Speedometer showed no issues.  (there were some differences in values, but all fell within the normal swings of the repective tests).

**Pull request merged**